src/dump.cpp

```diff
@@ -288,6 +288,102 @@ bool game::dump_stats( const std::string& what, dump_mode mode, const std::vecto
             }
         }
 
+    } else if( what == "GUN_DAMAGE" ) {
+        scol = -1; // unsorted output so graph columns have predictable ordering
+
+        header = { "Name" };
+        for( int range = 1; range <= MAX_RANGE; ++range ) {
+            header.push_back( to_string( range ) );
+        }
+
+        auto dump = [&rows]( const standard_npc & who, const item & gun, int aim_moves ) {
+            double recoil = MIN_RECOIL;
+            for( int i = 0; i < aim_moves; ++i ) {
+                double adj = who.aim_per_move( gun, recoil );
+                if( adj <= 0 ) {
+                    aim_moves = i;
+                    break;
+                }
+                recoil -= adj;
+            }
+
+            double dispersion = who.get_weapon_dispersion( gun ) + recoil;
+
+            std::vector<std::string> r;
+            r.push_back( string_format( "%s %s (%dmv aiming)", who.get_name().c_str(), gun.tname().c_str(), aim_moves ) );
+
+            for( int range = 1; range <= MAX_RANGE; ++range ) {
+                if( range > gun.gun_range( &who ) ) {
+                    r.push_back( "0" );
+                    continue;
+                }
+
```

Comment:
Ok, I'll add some more comments to that bit of the code.

It is finding the mean (expected value) of the damage multiplier for a given dispersion. It does that by defining a probability density function (defined in the comments below) for the damage multiplier, which depends on the various hit chances, and then integrating x.f(x). See https://en.wikipedia.org/wiki/Expected_value section 4 (univariate continuous random variable). It actually could be collapsed right down to just a few magic constants rather than having the xfx_integral lambda, but I think it's clearer to have the whole definition visible.